python/ray/_private/ray_constants.py

```diff
@@ -480,3 +480,5 @@ def gcs_actor_scheduling_enabled():
 
 
 RAY_NODE_IP_FILENAME = "node_ip_address.json"
+
+RAY_ENABLE_MIG_DETECTION_ENV_VAR = "RAY_ENABLE_MIG_DETECTION"
```

Comment:
Let's do MIG support in a separate PR. For this PR, let's just remove gpustats and GPUtil and replace them with pynvml.

Reply:
ohh this is just to enable mig instance detection, for mig feature later, it will be a flag `RAY_ENABLE_MIG_ENV_VAR`

Reply:
Yea I'm suggesting we remove all mig related changes from this PR and just focus on removing gpustat and GPUtil dependencies. This should already be a big enough change for this PR since you also need to fix dashboard.

Reply:
I planned to split the change with auto detect and remove gpustat PRs instead since one cover entirely core while other is entirely dashboard
https://github.com/ray-project/ray/pull/41044 for the gpustat pr